libs/dh/wholesale/feature-settlement-reports/src/lib/table/dh-settlement-reports-table.component.ts

```diff
@@ -117,25 +119,20 @@ export class DhSettlementReportsTableComponent {
       return;
     }
 
-    const fileName = this.settlementReportName(settlementReport);
-    const fileOptions = { name: fileName, type: 'application/zip' };
+    settlementReportDownloadUrl = `${settlementReportDownloadUrl}&filename=${this.settlementReportName(settlementReport)}&fromApi=${settlementReport.fromApi}`;
 
-    this.toastService.open({
-      type: 'loading',
-      message: translate('shared.downloadStart'),
+    const result = await this.addTokenToDownloadUrlMutation.mutate({
+      variables: { url: settlementReportDownloadUrl },
     });
 
-    this.httpClient
-      .get(settlementReportDownloadUrl, { responseType: 'blob' })
-      .pipe(switchMap(streamToFile(fileOptions)))
-      .subscribe({
-        complete: () => this.toastService.dismiss(),
-        error: () =>
-          this.toastService.open({
-            type: 'danger',
-            message: translate('shared.downloadFailed'),
-          }),
-      });
+    const downloadUrl = result.data?.addTokenToDownloadUrl.downloadUrlWithToken;
+
+    if (downloadUrl) {
+      const link = document.createElement('a');
+      link.href = downloadUrl;
+      link.target = '_blank';
+      link.click();
```

Comment:
suggestion: let's remove the element after the click


```suggestion
      link.click();
      link.remove();
```